src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,19 +28,31 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
-        this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
+        String[] splittedAddress = address.split(",",0);
+        if(splittedAddress.length != 4){
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+        initialiseAddress(isPrivate, splittedAddress);
+//        if (!isValidAddress(trimmedAddress)) {
+//            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+//        }
         this.value = trimmedAddress;
     }
+    /**Initialises the address with its respective members */
+    private void initialiseAddress(boolean isPrivate, String[] splittedAddress) {
+        this.blockNo = new Block(splittedAddress[0]);
+        this.street = new Street(splittedAddress[1]);
+        this.unit = new Unit(splittedAddress[2]);
+        this.postalCode = new PostalCode(splittedAddress[3]);
+        this.isPrivate = isPrivate;
+    }
 
     /**
      * Returns true if a given string is a valid person address.
      */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
-    }
+//    public static boolean isValidAddress(String test) {
+//        return test.matches(ADDRESS_VALIDATION_REGEX);
+//    }
```

Comment:
You can consider adding validation for individual fields, e.g. postal code must be 6-digit number.